src/translate.py

```diff
@@ -3823,7 +3833,7 @@ def for_type(type: Type, array_dim: Optional[int]) -> Optional[str]:
                     if el is None:
                         return None
                     elements.append(el)
-                return f"{{{', '.join(elements)}}}"
+                return fmt.format_array(elements)
```

Comment:
I was able to simplify `get_ctype_fields()`, but reducing the similar code here is harder:
- I tried to keep it so that `translate.py` mostly deals with `Type`s and doesn't construct/introspect `CType`s -- I think it's good to keep that boundary?
- Since arrays can only be represented by CTypes, it means that this would involve a lot of roundtripping between Type -> CType -> Type -> ...

Reply:
Could add helper for constructing an array to types.py without too much type knowledge in translate.py, but yeah your second point is fair. Let's skip this.